src/server/scripts/Commands/cs_rbac.cpp

```diff
@@ -32,349 +32,256 @@ EndScriptData */
+    ChatCommandTable GetCommands() const override
     {
-        static std::vector<ChatCommand> rbacAccountCommandTable =
+        static ChatCommandTable rbacAccountCommandTable =
         {
-            {        "list", rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_LIST,   true, &HandleRBACPermListCommand,    "" },
-            {       "grant", rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_GRANT,  true, &HandleRBACPermGrantCommand,   "" },
-            {        "deny", rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_DENY,   true, &HandleRBACPermDenyCommand,    "" },
-            {      "revoke", rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_REVOKE, true, &HandleRBACPermRevokeCommand,  "" },
+            { "list",   HandleRBACPermListCommand,   rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_LIST,   Console::Yes },
+            { "grant",  HandleRBACPermGrantCommand,  rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_GRANT,  Console::Yes },
+            { "deny",   HandleRBACPermDenyCommand,   rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_DENY,   Console::Yes },
+            { "revoke", HandleRBACPermRevokeCommand, rbac::RBAC_PERM_COMMAND_RBAC_ACC_PERM_REVOKE, Console::Yes },
         };
 
-        static std::vector<ChatCommand> rbacCommandTable =
+        static ChatCommandTable rbacCommandTable =
         {
-            {    "account", rbac::RBAC_PERM_COMMAND_RBAC_ACC,  true, nullptr, "", rbacAccountCommandTable },
-            {       "list", rbac::RBAC_PERM_COMMAND_RBAC_LIST, true, &HandleRBACListPermissionsCommand, "" },
+            { "account", rbacAccountCommandTable },
+            { "list",    HandleRBACListPermissionsCommand, rbac::RBAC_PERM_COMMAND_RBAC_LIST, Console::Yes },
         };
 
-        static std::vector<ChatCommand> commandTable =
+        static ChatCommandTable commandTable =
         {
-            {       "rbac", rbac::RBAC_PERM_COMMAND_RBAC, true, nullptr, "", rbacCommandTable },
+            { "rbac", rbacCommandTable },
         };
 
         return commandTable;
     }
 
-    static RBACCommandData* ReadParams(ChatHandler* handler, char const* args, bool checkParams = true)
+    static RBACCommandData GetRBACData(AccountIdentifier account)
     {
-        if (!args)
-            return nullptr;
-
-        char* param1 = strtok((char*)args, " ");
-        char* param2 = strtok(nullptr, " ");
-        char* param3 = strtok(nullptr, " ");
-
-        int32 realmId = -1;
-        uint32 accountId = 0;
-        std::string accountName;
-        uint32 id = 0;
-        RBACCommandData* data = nullptr;
-        rbac::RBACData* rdata = nullptr;
-        bool useSelectedPlayer = false;
-
-        if (checkParams)
-        {
-            if (!param3)
-            {
-                if (param2)
-                    realmId = atoi(param2);
-
-                if (param1)
-                    id = atoi(param1);
-
-                useSelectedPlayer = true;
-            }
-            else
-            {
-                id = atoi(param2);
-                realmId = atoi(param3);
```

Comment:
What is/was `param1` when this branch is taken?

Reply:
The account name e.g `rbac acc rev acc1 2 3` is account name "acc1", permission ID 2 and realm 3